src/main/scala/commonist/util/OaiPmh2.scala

```diff
@@ -0,0 +1,122 @@
+package commonist.util
+
+import commonist.data._
+
+import scala.xml._
+
+import scutil.log._
+
+/** 
+ * Parse and extract image metadata from OAI-PMH 2.0 files.
+ * See www.openarchives.org/OAI/openarchivesprotocol.html
+ */
+class OaiPmh2(doc:Elem,props:Map[String,String]) extends Logging {
+
+  /** fix badly formatted XML files (escaped twice) **/
+  private def text(e:NodeSeq) = {
+    e.map(_.text).mkString("\n").trim()
+     .replaceAll("&amp;", "&").replaceAll("&quot;", "\"").replaceAll("&apos;", "'").replaceAll("&lt;", "<").replaceAll("&gt;", ">")
+  }
+
+  private def artist(creator:String):String = {
+    val prop = creator.replaceAll(" ", "_")
+    if (props.contains(prop))
+      return props(prop)
+    return creator
+  }
+
+  private def permission(rights:String):String = {
+    val lowercase = rights.toLowerCase()
+    if (lowercase.contains("public domain")
```

Comment:
early return is frowned on in scala, just use `if (...) "{{PD-old|PD-70}}" else rights`